### PR TITLE
minor fix and improvements on localkeystore

### PR DESCRIPTION
**Pull request by gnunicorn**
This removes `pub` a bunch of internal methods of `LocalKeystore`, adds a test for local-keystore to only hold but then drop keys generated with a seed (as the trait docs define it) and then a fix that actually does that.

separated from ##7365

### CODE REVIEW

File: client/keystore/src/local.rs [hunk 1]
```diff
@@ -386,20 +386,22 @@ impl KeystoreInner {
 	/// Generate a new key.
 	///
 	/// Places it into the file system store.
```
Comment (bkchr) [inline client/keystore/src/local.rs:388]: Docs outdated.

**Review by bkchr — APPROVED** [review]

**Review by wheresaddie — APPROVED** [review]
thanks @gnunicorn

File: client/keystore/src/local.rs [hunk 2]
```diff
@@ -501,6 +503,8 @@ mod tests {
 		str::FromStr,
 	};
 
+	const TEST_KEY_TYPE: KeyTypeId = KeyTypeId(*b"test");
```
Comment (rakanalh) [inline client/keystore/src/local.rs:506]: Nitpick: i prefer constants to be declared at the top instead of in-between implementations

**Review by rakanalh — APPROVED** [review]
LGTM

File: client/keystore/src/local.rs (hunk 2 above)
Comment (bkchr) [reply client/keystore/src/local.rs:506]: This is not between any implementations. It is at the top of the tests module.

**Comment by gnunicorn** [comment]
bot merge

**Comment by gnunicorn** [comment]
bot merge

**Pull request merged**